### DISCUSSION TITLE
Wolframite development plans and event log 2023-2025

### DISCUSSION THREAD

**Discussion by holyjak** [scicloj/wolframite#17] [Announcements]
This discussion tracks key events and plans in the live of Wolframite, so that interested parties can keep an eye on what is going on.

**Comment by holyjak** [comment]
# 2023-11-23 Reviving Wolframite 

Me and Thomas (@light-matters) had a chat about reviving Wolframite and bringing it to a usable state - primarily understanding and bringing to a conlusion the work Pawel (@qazwsxpawel, or prnc at Zulip) has started on cleaning up the API and integration in the `develop` branch, and making sure that docs are great.

We wil look at what he did in the `develop` branch, and make a plan for bringing it to master, as well as updating the docs accordingly.

Further updates will be posted here, while some more transient discussions may be hapening in the [wolfram-clojure-bridge](https://clojurians.zulipchat.com/#narrow/stream/313853-wolfram-clojure-bridge) Zulip stream.

Tips for getting familiar with the code and bringing it further:

* go through the explainer and some other demo nss to get a feel how things work
* construct a test suite from wolfram docs + the issues uncovered (see above)
* Review [recordings from meetings/sessions](https://clojurians.zulipchat.com/#narrow/stream/313853-wolfram-clojure-bridge/topic/recordings) the last time Wolframite was worked on

**Reply by holyjak** [reply]
# 2025-01-17 v1 released

We have released v1 and are looking into promoting it. The code is rather stable, though we expect to evolve it based on user feedback.

**Reply by abhi18av** [reply]
This is amazing news - thank you for all your hard work on this @holyjak ! 

I am planning a couple of blog posts based on this package related to Bioinformatics.

**Reply by holyjak** [reply]
Exciting!

**Comment by holyjak** [comment]
2023-11-24

I have created a project board to track the most important tasks. Some documentation work started in #18

**Comment by light-matters** [comment]
Feature ideas:
- Create a bridging module so that 'Emmy' equations can be passed to Wolfram and vice-versa.
- Make it easy to add your own symbol aliases

**Reply by holyjak** [reply]
I've turned this into https://github.com/scicloj/wolframite/discussions/19, https://github.com/scicloj/wolframite/discussions/20

**Comment by holyjak** [comment]
# 2023-12-01 Status update: Code looks good, wrapping up docs & merging soon

The code on the dev branch looks good, I want to finish reviewing it and adding docs (which I started in #18) and merge all of it to master.

An open question is whether `custom-parse` should require a flag to actually work (current state) or not (my preferred).

Me and @light-matters will meet next week to use latest Wolframite on a real project, to see how that goes.

Within a few weeks, I should wrap up the two open PRs, and update docs in README, and hopefully rewrite `load-all-syms` into a much faster one. Then we can start working on other kinds of docs - intro, tutorial, ...; likely inspired by old Clojuratica ones.

**Comment by holyjak** [comment]
# 2023-12-10

I have merged pending PRs into master, renamed it to main, and made it the repo's primary branch.

Also created #24 

Next plans: Make finding/loading jlink.jar more robust. Update Readme.

**Comment by holyjak** [comment]
# 2024-01-19

We have brought the README up to date, drop the need for `:custom-parse` flag, made jlink detection better and more robust.

Plans now: rework load-all-symbols to create vars with symbols instead of callable functions, which turned out to be problematic. Learn and document the code more, perhaps add some more tests. Reach out to Wolfram with our questions on better errors and performance.

**Comment by holyjak** [comment]
# 2024-02-22

We've planned getting to a release and agreed on making wolframite-1.0.0-alpha, then seek feedback from the community. We likely want to prepare some learning materials beforehand, so it is easier for folks to actually get started.

We have agreed to focus on two target groups, Clojure devs interested in data science, and folks who know Mathematica / Wolfram and would benefit from using it from Clojure.

[Issues that should be included in v1 are now marked as such](https://github.com/scicloj/wolframite/issues?q=is%3Aopen+is%3Aissue+milestone%3Av1), and we've added few docs issues.

**Comment by holyjak** [comment]
# 2024-05-31 Great progress

I having been good at posting updates lately but we are making very good progress.  Thomas is using Wolframite on a real project, finding limitations and possible improvements we are then implementing. JLink detection is much more stable, and we have almost finished switching to an explicit initialization. You can see the key issues to fix before v1 in here https://github.com/scicloj/wolframite/milestone/1

I also plan very soon to start writing documents to gather input from Wolfram Engineering.

**Comment by holyjak** [comment]
# 2024-06-28

We have reviewed the almost finished #55, which fixes 3 issues and is the last key part of stabilizing the api. Then we discussed testing of this, and plans to review what we promised to Clojurians Together vs. what we have done, and agreed on a date to plan more the tutorials we want to write.

**Comment by holyjak** [comment]
# 2024-08-02

Quick update: Our Clojurists Together sponsorship is at its end. We have done a couple of improvement to the code and ergonomics of using Wolframite, and Thomas has put a lot of effort into our [new documentation site](https://scicloj.github.io/wolframite/). And documentation is something we still focus on.

**Comment by holyjak** [comment]
# 2024-09-27

We have written most of the documentation and made all the code changes we wanted. We just need to do a deep review of the docs, and I'm still expanding the demo [Using Wolframite to analyse cycle trips data](https://scicloj.github.io/wolframite/for_developers.demo_analysis_cycling.html). This teaches me a lot about Wolfram and Wolframite and thus may spur further docs and code improvements, but nothing major.

**Reply by abhi18av** [reply]
Thanks for all this work and documentation @holyjak !
